GitUI/CommandsDialogs/BrowseDialog/DashboardControl/Dashboard.cs

```diff
@@ -372,7 +177,11 @@ private void groupLayoutPanel_DragDrop(object sender, DragEventArgs e)
                         }
                     }
 
-                    Repositories.AddMostRecentRepository(module.WorkingDir);
+                    ThreadHelper.JoinableTaskFactory.RunAsync(async () =>
```

Comment:
❓ Why is this using `RunAsync` rather than `Run`?

Reply:
So it can be FnF. By the time a new repository gets opened the underlying history gets updated in the storage.
Do you think it should run synchronously?

Reply:
> Do you think it should run synchronously?

At this location, synchronously would be easier to analyze since it follows the current behavior pattern. This has a clear advantage in building confidence that the PR contains working code. The suggestion depends on the impact:

* If synchronously is typically not noticeable, we could just use that and leave it for now.
* If synchronously is noticeable, but not "substantially worse than current" (e.g. contributing to *several* seconds new delays relative to current code), we can use synchronous to expedite this review but file a bug to follow up in a review after
* If synchronously here is substantially worse than current, it could switch to asynchronous and absorb the cost in the current review.